<html lang="auto">
 <head>
  <meta http-equiv="X-Translated-By" content="Google">
  <meta http-equiv="X-Translated-To" content="zh-CN">
  <script type="text/javascript" src="https://www.gstatic.com/_/translate_http/_/js/k=translate_http.tr.zh_CN.VYDj-nYYJwk.O/d=1/rs=AN8SPfq4Fg0mdeehsroELRJoc51-OEMCyQ/m=ajaxproxy" data-sourceurl="http://dwm.suckless.org/patches/actualfullscreen/dwm-actualfullscreen-20211013-cb3f58a.diff"></script>
  <meta name="robots" content="none">
 </head>
 <body>
  <script type="text/javascript" src="https://www.gstatic.com/_/translate_http/_/js/k=translate_http.tr.zh_CN.VYDj-nYYJwk.O/d=1/exm=ajaxproxy,el_conf/ed=1/rs=AN8SPfq4Fg0mdeehsroELRJoc51-OEMCyQ/m=navigationui" data-environment="prod" data-proxy-url="https://dwm-suckless-org.translate.goog" data-proxy-full-url="https://dwm-suckless-org.translate.goog/patches/actualfullscreen/dwm-actualfullscreen-20211013-cb3f58a.diff?_x_tr_sl=auto&amp;_x_tr_tl=zh-CN&amp;_x_tr_hl=zh-CN&amp;_x_tr_sch=http" data-source-url="http://dwm.suckless.org/patches/actualfullscreen/dwm-actualfullscreen-20211013-cb3f58a.diff" data-source-language="auto" data-target-language="zh-CN" data-display-language="zh-CN" data-detected-source-language="en" data-is-source-untranslated="false" data-source-untranslated-url="https://translate.google.com/website?sl=auto&amp;tl=zh-CN&amp;hl=zh-CN&amp;u=http://dwm.suckless.org/patches/actualfullscreen/dwm-actualfullscreen-20211013-cb3f58a.diff&amp;anno=2"></script>
  <pre>From eea13010ffc3983392857ee1e3804e3aa1064d7a Mon Sep 17 00:00:00 2001
From: Soenke Lambert 
   <s.lambert@mittwald.de>
    
Date: Wed, 13 Oct 2021 18:21:09 +0200
Subject: [PATCH] Fullscreen current window with [Alt]+[Shift]+[f]

This actually fullscreens a window, instead of just hiding the statusbar
and applying the monocle layout.
---
 config.def.h | 1 +
 dwm.1        | 3 +++
 dwm.c        | 8 ++++++++
 3 files changed, 12 insertions(+)

diff --git a/config.def.h b/config.def.h
index 1c0b587..8cd3204 100644
--- a/config.def.h
+++ b/config.def.h
@@ -78,6 +78,7 @@ static Key keys[] = {
 	{ MODKEY,                       XK_m,      setlayout,      {.v = &amp;layouts[2]} },
 	{ MODKEY,                       XK_space,  setlayout,      {0} },
 	{ MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
+	{ MODKEY|ShiftMask,             XK_f,      togglefullscr,  {0} },
 	{ MODKEY,                       XK_0,      view,           {.ui = ~0 } },
 	{ MODKEY|ShiftMask,             XK_0,      tag,            {.ui = ~0 } },
 	{ MODKEY,                       XK_comma,  focusmon,       {.i = -1 } },
diff --git a/dwm.1 b/dwm.1
index 13b3729..a368d05 100644
--- a/dwm.1
+++ b/dwm.1
@@ -116,6 +116,9 @@ Zooms/cycles focused window to/from master area (tiled layouts only).
 .B Mod1\-Shift\-c
 Close focused window.
 .TP
+.B Mod1\-Shift\-f
+Toggle fullscreen for focused window.
+.TP
 .B Mod1\-Shift\-space
 Toggle focused window between tiled and floating state.
 .TP
diff --git a/dwm.c b/dwm.c
index 4465af1..c1b899a 100644
--- a/dwm.c
+++ b/dwm.c
@@ -211,6 +211,7 @@ static void tagmon(const Arg *arg);
 static void tile(Monitor *);
 static void togglebar(const Arg *arg);
 static void togglefloating(const Arg *arg);
+static void togglefullscr(const Arg *arg);
 static void toggletag(const Arg *arg);
 static void toggleview(const Arg *arg);
 static void unfocus(Client *c, int setfocus);
@@ -1719,6 +1720,13 @@ togglefloating(const Arg *arg)
 	arrange(selmon);
 }
 
+void
+togglefullscr(const Arg *arg)
+{
+  if(selmon-&gt;sel)
+    setfullscreen(selmon-&gt;sel, !selmon-&gt;sel-&gt;isfullscreen);
+}
+
 void
 toggletag(const Arg *arg)
 {
-- 
2.30.2


   </s.lambert@mittwald.de></pre>
  <script>function gtElInit() {var lib = new google.translate.TranslateService();lib.translatePage('en', 'zh-CN', function () {});}</script>
  <script src="https://translate.google.com/translate_a/element.js?cb=gtElInit&amp;hl=zh-CN&amp;client=wt" type="text/javascript"></script>
 </body>
</html>